### ISSUE TITLE
serverless [service]:package fails if another service refers as params

### ISSUE THREAD

**Issue by mokuzuu** [serverless/compose#148]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Are you using the latest version?

- [X] Yes, I'm using the latest version

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

When service-b refers the output of service-a as params in serverless-compose.yml, packaging service-a fails.

e.g.
I have the following serverless-compose.yml
```
services:
  service-a:
    path: service-a 
  service-b:
    path: service-b
    params:
      value1: ${service-a.value1}
```
I ran the following command
```
npm run sls -- service-a:package

// package.json
"scripts": {
  "sls": "serverless"
}
```
I get following error
```
Error:
The variable "${service-a.value1}" cannot be resolved: the referenced output does not exist.
```

### Service configuration (serverless-compose.yml) content

```yaml
services:
  service-a:
    path: service-a
  service-b:
    path: service-b
    params:
      value1: ${service-a.value1}
      value2: ${service-a.value2}
```

### Command name and used flags

npm run sls -- service-a:package --stage [stage-name]  // package.json "scripts": {   "sls": "serverless" }

### Command output

```shell
> package-name@0.0.1 sls
> serverless "service-a:package" "--stage" "[stage-name]"

Environment: darwin, node 16.15.1, compose 1.2.4
Docs:        slss.io/docs-compose
Bugs:        github.com/serverless/compose/issues

Error:
The variable "${service-1.value1}" cannot be resolved: the referenced output does not exist.

Verbose logs are available in ".serverless/compose.log"
```

**Comment by mokuzuu** [comment]
Actually, deployment for `service-a` also fails but I feel it shouldn't...?

e.g.
`npm run sls -- service-a:deploy`

**Comment by mnapoli** [comment]
Could it be that the service has been deployed from another machine? (https://www.serverless.com/framework/docs/guides/compose#refreshing-outputs)

You might want to refresh the outputs stored locally with:

```
serverless refresh-outputs
```

**Comment by mokuzuu** [comment]
Sorry for the late reply. I ran the command below but it didn't work...
```shell
npm run sls:refresh -- --stage [stage-name] // sls:refresh: "serverless refresh-outputs"

> project@0.0.1 sls:refresh
> serverless refresh-outputs "--stage" "[stage-name]"


Refreshing outputs

    ✔  service-a › outputs refreshed › 3s
    ✔  service-b › outputs refreshed › 4s

npm run sls:deploy -- service-a:deploy -- --stage [stage-name] // sls:deploy: "serverless deploy"

> project@0.0.1 sls:deploy
> serverless deploy "service-a:deploy"

Environment: darwin, node 16.15.1, compose 1.2.4
Docs:        slss.io/docs-compose
Bugs:        github.com/serverless/compose/issues

Error:
The variable "${service-a.value1}" cannot be resolved: the referenced output does not exist.

Verbose logs are available in ".serverless/compose.log"
```

**Comment by mnapoli** [comment]
Thanks. Could it be that the output simply doesn't exist? Could you run the command `sls outputs`?

**Comment by semiautomatixfu** [comment]
This is a bit of a chicken-and-egg situation for `package` where a service has params.

```
$ ./node_modules/serverless/bin/serverless.js package --service member --stage dev
Environment: linux, node 14.19.2, compose 1.2.4
Docs:        slss.io/docs-compose
Bugs:        github.com/serverless/compose/issues

Error:
The variable "${shared.checkoutDataTableName}" cannot be resolved: the referenced output does not exist.

Verbose logs are available in ".serverless/compose.log"
```

```
$ ./node_modules/serverless/bin/serverless.js package --service member --stage dev --param checkoutDataTableName="" --param checkoutDataTableArn=""
(node:87066) UnhandledPromiseRejectionWarning: ServerlessError: The "--param" option is not supported (yet) in Serverless Framework Compose
You can search and/or open feature requests here: https://github.com/serverless/compose'
    at /home/tim/Documents/Git/api.checkout.fu/node_modules/@serverless/compose/src/validate-options.js:12:13
    at Array.forEach (<anonymous>)
    at validateCliOptions (/home/tim/Documents/Git/api.checkout.fu/node_modules/@serverless/compose/src/validate-options.js:10:31)
    at runComponents (/home/tim/Documents/Git/api.checkout.fu/node_modules/@serverless/compose/src/index.js:125:3)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:87066) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:87066) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

tl;dr if we define params in the `serverless.yml` file, then it is neither possible to use `package` with or without specifying the params.

```yml
services:
  service-a:
    path: service-a
  service-b:
    path: service-b
    params:
      value1: ${service-a.value1}
      value2: ${service-a.value2}
```

**Comment by semiautomatixfu** [comment]
@pgrzesik  Additionally we're running `dynamodb-local` and `offline` to add to the issues.

```
$ ./node_modules/serverless/bin/serverless.js package --service service-a --stage ${USER} --verbose
Environment: linux, node 14.19.2, compose 1.2.4
Docs:        slss.io/docs-compose
Bugs:        github.com/serverless/compose/issues

Error:
The variable "${service-a.value1}" cannot be resolved: the referenced output does not exist.
ServerlessError: The variable "${service-a.value1}" cannot be resolved: the referenced output does not exist.
    at Object.<anonymous> (/node_modules/@serverless/compose/src/ComponentsService.js:45:15)
    at walker (/node_modules/traverse/index.js:190:22)
    at /node_modules/traverse/index.js:208:29
    at Array.forEach (<anonymous>)
    at forEach (/node_modules/traverse/index.js:298:31)
    at walker (/node_modules/traverse/index.js:203:13)
    at /node_modules/traverse/index.js:208:29
    at Array.forEach (<anonymous>)
    at forEach (/node_modules/traverse/index.js:298:31)
    at walker (/node_modules/traverse/index.js:203:13)

Verbose logs are available in ".serverless/compose.log"
$ ./node_modules/serverless/bin/serverless.js refresh-outputs --verbose

Refreshing outputs
service-a › waiting
member › waiting
service-a › refreshing outputs
service-a › Running "serverless info --verbose --stage tim"
service-a › error
service-a › spawn serverless ENOENT
service-a › Error: spawn serverless ENOENT
service-a ›     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:19)
service-a ›     at onErrorNT (internal/child_process.js:472:16)
service-a ›     at processTicksAndRejections (internal/process/task_queues.js:82:21)
member › skipped

    ✖  service-a › error › 0s

spawn serverless ENOENT

       member › skipped

Verbose logs are available in ".serverless/compose.log"
```

**Comment by pgrzesik** [comment]
That's very surprising - do you have a chance to provide a small reproducible example that I could run on my side to evaluate these issues? I couldn't reproduce it with my test services

**Comment by tibbe** [comment]
Some more testing suggests that `sls package`/`sls refresh-outputs` only works with a previously successfully deployed stage.

**Comment by darylteo** [comment]
For those looking for a workaround on this (similar to my suggested proposal here https://github.com/serverless/compose/issues/149) , I ended up creating 2 serverless-compose.{stage}.yml, and then manually copying them to serverless-compose.yml in as `package.json` scripts.
